frontend/test/pytest/test_mitigation.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2022-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test integration for catalyst.mitigate_with_zne."""
+
+import jax
+import numpy as np
+import pennylane as qml
+import pytest
+
+import catalyst
+
+
+@pytest.mark.parametrize("params", [0.1, 0.2, 0.3, 0.4, 0.5])
+def test_single_measurement(params):
+    """Test that without noise the same results are returned for single measurements."""
+    dev = qml.device("lightning.qubit", wires=2)
+
+    @qml.qnode(device=dev)
+    def circuit(x):
+        qml.Hadamard(wires=0)
+        qml.RZ(x, wires=0)
+        qml.RZ(x, wires=0)
+        qml.CNOT(wires=[1, 0])
+        qml.Hadamard(wires=1)
+        return qml.expval(qml.PauliY(wires=0))
+
+    @catalyst.qjit
+    def mitigated_qnode(args):
+        return catalyst.mitigate_with_zne(circuit, scale_factors=jax.numpy.array([1, 2, 3]), deg=2)(
```

Comment:
Can scale factors be different from `[1, 2, 3]`? Other `deg`?

Reply:
Addendum, can we calculate the `deg` from the scale factor. Example given, `deg = len(scale_factors) - 1`?

Reply:
Scale factors is an array of int and deg is best `deg = len(scale_factors) - 1` that's why I added it as default.